src/app/screens/Accounts/NostrSettings/index.tsx

```diff
@@ -88,38 +67,56 @@ function NostrSettings() {
 
   // TODO: simplify this method - would be good to have a dedicated "remove nostr key" button
   async function handleSaveNostrPrivateKey() {
-    if (
-      currentPrivateKey &&
-      prompt(
-        t("nostr.private_key.warning", { name: account?.name })
-      )?.toLowerCase() !== account?.name?.toLowerCase()
-    ) {
-      toast.error(t("nostr.private_key.failed_to_remove"));
-      return;
-    }
-
     try {
-      if (nostrPrivateKey) {
-        await api.nostr.setPrivateKey(id, nostrPrivateKey);
-      } else {
-        await api.nostr.removePrivateKey(id);
+      // TODO: is there a way this can be moved to the background script and use the Nostr object?
+      // NOTE: it is done this way to show the user the new public key before saving
+      setNostrPublicKey(
```

Comment:
this doesn't work - the nostrPublicKey is now blank until you click save.

What about we undo the changes and just not show an error toast if we fail to set the pubkey?

Reply:
Okay then we will just set the public key to an empty string if the input private key is not valid and then when after the user clicks on the save button then we can validate, if the public key does not exist then we can error toast it.
Does this method sound good?

Reply:
@Rithvik-padma sorry, I think this actually can be fixed by a minor change of changing a toast to logging the error to the console. We already have error handling when we save the key.